### PR TITLE
Vinh test coverage

### PR DESCRIPTION
**Pull request by dlvinh**
Test coverage on Login Screen

### CODE REVIEW

**Comment by dlvinh** [comment]
image

File: lib/Screen/HomeScreen.dart [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                          child: const Text(
+                            "Continue",
+                            style: TextStyle(color: Colors.black, fontSize: 20),
+                          )),
+                      const SizedBox(
+                        height: 10,
+                      ),
+                      ElevatedButton(
+                        child: const Text.rich(
+                          TextSpan(children: [
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:140]: Great choice of widget use

File: lib/Screen/HomeScreen.dart [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                )),
+                Container(
+                  padding: const EdgeInsets.fromLTRB(30, 0, 30, 0),
+                  // color:Colors.amberAccent,
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.stretch,
+                    children: <Widget>[
+                      ElevatedButton(
+                          style: ButtonStyle(
+                            backgroundColor: !_isButtonDisable
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:119]: You can use MaterialStateProperty.resolveWith()

Reference: https://api.flutter.dev/flutter/material/MaterialStateProperty-class.html

How to use:
- remove bool condition to check button state
- use MaterialStateProperty.resolveWith() to switch colour between different states
- button is disabled when onPressed is null

File: lib/Screen/HomeScreen.dart [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                                ? MaterialStateProperty.all<Color>(
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:120]: ElevatedButtonTheme is already defined in ThemeData, you don't need to override it here unless you wanna use different theme

File: lib/Screen/HomeScreen.dart [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                          ),
+                          onPressed: () {
+                            Navigator.push(context,
+                                MaterialPageRoute(builder: (context) {
+                              return VerificationPage(emailValue: emailValue);
+                            }));
+                          },
+                          child: const Text(
+                            "Continue",
+                            style: TextStyle(color: Colors.black, fontSize: 20),
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:133]: TextStyle fit elevated button can be set in ElevatedButtonTheme

**Review by suesitran — CHANGES_REQUESTED** [review]
Just a few comments

File: lib/Screen/HomeScreen.dart [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+  String emailValue = "";
+
+  @override
+  Widget build(BuildContext context) {
+    return Theme(
+        data: ThemeData(
+            elevatedButtonTheme: ElevatedButtonThemeData(
+                style: ButtonStyle(
+                  fixedSize: MaterialStateProperty.all<Size>(
+            const Size.fromHeight(45),
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:21]: `Size` with fixed height is not recommended around `Text`

File: lib/Screen/HomeScreen.dart [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+          ),
+          backgroundColor: MaterialStateProperty.all<Color>(
+              const Color.fromRGBO(255, 255, 255, 1)),
+            ))),
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:25]: use this `MaterialStateProperty` to set different color for different button state

File: lib/Screen/HomeScreen.dart [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                    // color:Colors.amberAccent,
+                    child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    const Text(
+                      "Space",
+                      style: TextStyle(
+                          color: Colors.white,
+                          fontSize: 65,
+                          fontWeight: FontWeight.bold),
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:55]: use `Theme.of(context).textTheme.headline2`, either healine2 or 1, need to check the size.

File: lib/Screen/HomeScreen.dart [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+                    ),
+                    Container(
+                      padding: const EdgeInsets.fromLTRB(0, 10, 0, 10),
+                      child: const Text(
+                        "Please enter your email address",
+                        style: TextStyle(
+                            color: Colors.white,
+                            fontSize: 20,
+                            fontWeight: FontWeight.w500),
+                      ),
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:65]: same as above

File: lib/Screen/HomeScreen.dart [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+                            color: Colors.white,
+                            fontWeight: FontWeight.w500,
+                            fontSize: 20),
+                        decoration: const InputDecoration(
+                          isCollapsed: true,
+                          contentPadding: EdgeInsets.all(10),
+                          filled: true,
+                          fillColor: Color.fromRGBO(255, 255, 255, 0.4),
+                          enabledBorder: OutlineInputBorder(
+                              borderRadius:
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:80]: good!

File: lib/Screen/HomeScreen.dart [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+                                    const Color.fromARGB(190, 255, 255, 255))
+                                : MaterialStateProperty.all<Color>(
+                                    const Color.fromRGBO(255, 255, 255, 1)),
+                          ),
+                          onPressed: () {
+                            Navigator.push(context,
+                                MaterialPageRoute(builder: (context) {
+                              return VerificationPage(emailValue: emailValue);
+                            }));
+                          },
```
Comment (suesitran) [inline lib/Screen/HomeScreen.dart:130]: _isButtonDisable ? null : () {}

File: pubspec.yaml [hunk 11]
```diff
@@ -38,7 +38,7 @@ dependencies:
 dev_dependencies:
   flutter_test:
     sdk: flutter
-
+  test_coverage: ^0.4.2
```
Comment (suesitran) [inline pubspec.yaml:41]: good!

File: test/widget_test/homescreen_test.dart [hunk 12]
```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hatspace/Screen/HomeScreen.dart';
+import 'package:hatspace/Screen/VerificationScreen.dart';
+
+import 'package:hatspace/main.dart';
+
+void homeScreenTest() {
```
Comment (suesitran) [inline test/widget_test/homescreen_test.dart:9]: void main() {}

**Review by suesitran — CHANGES_REQUESTED** [review]
please check comments

**Comment by dlvinh** [comment]
image

**Pull request closed without merging**